pkg/daemon/daemon.go

```diff
@@ -521,22 +500,47 @@ func (dn *Daemon) BindPodMounts() error {
 }
 
 func (dn *Daemon) runLoginMonitor(stopCh <-chan struct{}, exitCh chan<- error) {
-	sessionNewCh := dn.loginClient.Subscribe("SessionNew")
-	for {
-		select {
-		case <-stopCh:
-			return
-		case msg, ok := <-sessionNewCh:
-			if !ok {
-				glog.V(4).Info("Not adding the ssh accessed annotation because the logind SessionNew channel is closed")
-				return
-			}
-			glog.Infof("Detected a new login session: %v", msg)
-			glog.Infof("Login access is discouraged! Applying annotation: %v", machineConfigDaemonSSHAccessAnnotationKey)
-			if err := dn.applySSHAccessedAnnotation(); err != nil {
-				exitCh <- err
-			}
-		}
+	cmd := exec.Command("journalctl", "-b", "-f", "-o", "cat", "MESSAGE_ID=" + sdMessageSessionStart)
+	stdout, err := cmd.StdoutPipe()
+	if err != nil {
+		exitCh <- err
+		return
+	}
+	if err := cmd.Start(); err != nil {
+		exitCh <- err
+		return
+	}
+	worker := make(chan struct{})
+ 	go func() {
```

Comment:
nevermind, I realized this goroutine and the extra channel isn't needed, just this is ok:

```diff
diff --git a/pkg/daemon/daemon.go b/pkg/daemon/daemon.go
index 279891f..e818bd0 100644
--- a/pkg/daemon/daemon.go
+++ b/pkg/daemon/daemon.go
@@ -414,7 +414,7 @@ const (
 
 // detectEarlySSHAccessesFromBoot taints the node if we find a login before the daemon started up.
 func (dn *Daemon) detectEarlySSHAccessesFromBoot() error {
-	journalOutput, err := exec.Command("journalctl", "-b", "-o", "cat", "MESSAGE_ID=" + sdMessageSessionStart).CombinedOutput()
+	journalOutput, err := exec.Command("journalctl", "-b", "-o", "cat", "MESSAGE_ID="+sdMessageSessionStart).CombinedOutput()
 	if err != nil {
 		return err
 	}
@@ -490,7 +490,7 @@ func (dn *Daemon) BindPodMounts() error {
 }
 
 func (dn *Daemon) runLoginMonitor(stopCh <-chan struct{}, exitCh chan<- error) {
-	cmd := exec.Command("journalctl", "-b", "-f", "-o", "cat", "MESSAGE_ID=" + sdMessageSessionStart)
+	cmd := exec.Command("journalctl", "-b", "-f", "-o", "cat", "MESSAGE_ID="+sdMessageSessionStart)
 	stdout, err := cmd.StdoutPipe()
 	if err != nil {
 		exitCh <- err
@@ -500,37 +500,30 @@ func (dn *Daemon) runLoginMonitor(stopCh <-chan struct{}, exitCh chan<- error) {
 		exitCh <- err
 		return
 	}
-	worker := make(chan struct{})
- 	go func() {
-		for {
-			select {
-			case <-worker:
- 				return
-			default:
-				buf := make([]byte, 1024)
-				l, err := stdout.Read(buf)
-				if err != nil {
-					if err == io.EOF {
-						return
-					}
-					exitCh <- err
+	for {
+		select {
+		case <-stopCh:
+			cmd.Process.Kill()
+			return
+		default:
+			buf := make([]byte, 1024)
+			l, err := stdout.Read(buf)
+			if err != nil {
+				if err == io.EOF {
 					return
 				}
-				if l > 0 {
-					line := strings.Split(string(buf), "\n")[0]
-					glog.Infof("Detected a new login session: %s", line)
-					glog.Infof("Login access is discouraged! Applying annotation: %v", machineConfigDaemonSSHAccessAnnotationKey)
-					if err := dn.applySSHAccessedAnnotation(); err != nil {
-						exitCh <- err
-					}
+				exitCh <- err
+				return
+			}
+			if l > 0 {
+				line := strings.Split(string(buf), "\n")[0]
+				glog.Infof("Detected a new login session: %s", line)
+				glog.Infof("Login access is discouraged! Applying annotation: %v", machineConfigDaemonSSHAccessAnnotationKey)
+				if err := dn.applySSHAccessedAnnotation(); err != nil {
+					exitCh <- err
 				}
- 			}
- 		}
- 	}()
-	select {
-	case <-stopCh:
-		close(worker)
-		cmd.Process.Kill()
+			}
+		}
 	}
 }
 ```

Reply:
OK...I will do this but in the end the ssh tainting stuff IMO can regress a little bit...there's a massive pile of stuff backlogged behind the rhel8 rebase and I don't want to have the journal tainting code being *perfect* blocking that.

Reply:
sure thing, I concur with that, I just want to leave traces that we need to fix this. Take my comments as nits when appropriate

Reply:
But briefly, I don't understand; we'll block the goroutine in the `stdout.Read` invocation and hence never check `stopCh` right?